avocado/utils/configure_network.py

```diff
@@ -35,168 +37,248 @@ class NWException(Exception):
+    def unset_ip(self):
+        """
+        Utility to unassign IP to Defined Interface
+        """
+        if distro.detect().name == 'rhel':
+            conf_file = "/etc/sysconfig/network-scripts/ifcfg-%s" % self.name
 
-    cmd = "ifdown %s" % interface
-    try:
-        process.system(cmd, sudo=True)
-        shutil.move(conf_file+".backup", conf_file)
+        if distro.detect().name == 'SuSE':
+            conf_file = "/etc/sysconfig/network/ifcfg-%s" % self.name
+
+        self.bring_down()
+        self._move_config_file("%s.backup" % conf_file, conf_file)
+
+    def ping_check(self, peer_ip, count, option=None, flood=False):
+        """
+        utility perform Ping operation on Peer IP address and return status
+        :param peer_ip :  Peer IP address
+        :param count   :  ping count
+        :param option  :  Default is None
+        :param flood   :  Default is False
+        """
+
+        cmd = "ping -I %s %s -c %s" % (self.name, peer_ip, count)
+        if flood:
+            cmd = "%s -f" % cmd
+        elif option:
+            cmd = "%s %s" % (cmd, option)
+        if process.system(cmd, shell=True, verbose=True,
+                          ignore_status=True) != 0:
+            return False
         return True
-    except Exception as ex:
-        raise NWException("ifdown fails: %s" % ex)
 
+    def set_mtu(self, mtu):
+        """
+        utility set mtu size to a interface and return status
+        :param mtu :  mtu size that meed to be set
+        :ptype : String
+        :return : return True / False in case of mtu able to set
+        """
+        cmd = "ip link set %s mtu %s" % (self.name, mtu)
 
-def ping_check(interface, peer_ip, count, option=None, flood=False):
-    """
-    Checks if the ping to peer works.
-    """
-    cmd = "ping -I %s %s -c %s" % (interface, peer_ip, count)
-    if flood is True:
-        cmd = "%s -f" % cmd
-    elif option is not None:
-        cmd = "%s %s" % (cmd, option)
-    if process.system(cmd, shell=True, verbose=True,
-                      ignore_status=True) != 0:
+        cmd_mtu = "ip add show %s" % self.name
+        mtu_value = ''
+        try:
+            if self.remote_session:
+                self.remote_session.cmd(cmd)
+                mtu_value = self.remote_session.cmd(
+                    cmd_mtu).stdout.decode("utf-8")
+            else:
+                process.system(cmd, shell=True)
+                mtu_value = process.system_output(
+                    cmd_mtu, shell=True).decode("utf-8")
+            if mtu in mtu_value and wait.wait_for(self.is_link_up, timeout=120, args=[self.name]):
+                return True
+        except Exception:  # pylint: disable=W0703
+            return False
         return False
-    return True
 
+    def get_link_status(self):
+        if self.remote_session:
+            return self.session.cmd("cat /sys/class/net/%s/operstate" % self.name).stdout.decode('utf-8')
```

Comment:
Why do you need decode here? I did a small test locally and it works without the decode()

Reply:
ah yes right it should, will adopt